webapp/src/main/java/com/box/l10n/mojito/service/branch/notification/BranchNotificationService.java

```diff
@@ -0,0 +1,384 @@
+    Branch branch = branchRepository.findById(branchId).orElse(null);
+    sendMissingScreenshotNotificationsForBranch(branchNotificationMessageSender, branch);
+  }
+
+  void sendNotificationsForBranch(Branch branch) {
+    Preconditions.checkNotNull(branch);
+    Preconditions.checkNotNull(branch.getNotifiers());
+
+    logger.debug("sendNotificationsForBranch: {} ({})", branch.getId(), branch.getName());
+
+    BranchNotificationInfo branchNotificationInfo = getBranchNotificationInfo(branch);
+
+    branch.getNotifiers().stream()
+        .map(
+            notifierId -> {
+              BranchNotificationMessageSender messageSender =
+                  branchNotificationMessageSenders.getById(notifierId);
+              if (messageSender == null) {
+                throw new RuntimeException(
+                    "Can't get BranchNotificationMessageSender for id: " + notifierId);
+              }
+              return messageSender;
+            })
+        .forEach(
+            branchNotificationMessageSender -> {
+              try {
+                sendNotificationsForBranchWithSender(
+                    branchNotificationMessageSender, branch, branchNotificationInfo);
+              } catch (Exception e) {
+                logger.error("Fail safe, error tracking is up to each sender", e);
+              }
+            });
+  }
+
+  void sendNotificationsForBranchWithSender(
+      BranchNotificationMessageSender branchNotificationMessageSender,
+      Branch branch,
+      BranchNotificationInfo branchNotificationInfo) {
+    String notifierId = branchNotificationMessageSender.getId();
+
+    logger.debug(
+        "sendNotificationsForBranch: {} ({} with sender: {})",
+        branch.getId(),
+        branch.getName(),
+        notifierId);
+    BranchNotification branchNotification = getOrCreateBranchNotification(branch, notifierId);
+
+    if (shouldSendNewMessage(branchNotification, branchNotificationInfo)) {
+      sendNewMessage(
+          branchNotificationMessageSender, branch, branchNotification, branchNotificationInfo);
+      scheduleMissingScreenshotNotificationsForBranch(branch, notifierId);
+    } else if (shouldSendUpdatedMessage(branchNotification, branchNotificationInfo)) {
+      sendUpdatedMessage(
+          branchNotificationMessageSender, branch, branchNotification, branchNotificationInfo);
+      scheduleMissingScreenshotNotificationsForBranch(branch, notifierId);
+    }
+
+    if (shouldSendTranslatedMessage(branch, branchNotification)) {
+      sendTranslatedMessage(branchNotificationMessageSender, branch, branchNotification);
+    }
+  }
+
+  /**
+   * Schedule to check/send notification for missing screenshot in 30 minutes from now.
+   *
+   * @param branch
+   * @param notifierId
+   */
+  void scheduleMissingScreenshotNotificationsForBranch(Branch branch, String notifierId) {
+    Date date = DateTime.now().plusMinutes(30).toDate();
```

Comment:
Should the time (30m) be configurable?

Reply:
could be, never discuss changing it though so not urgent